pkg/mcoproducer/mcoproducer.go

```diff
@@ -0,0 +1,146 @@
+package mcoproducer
+
+import (
+	"encoding/base64"
+	"fmt"
+
+	"github.com/google/go-containerregistry/pkg/name"
+	"github.com/rh-ecosystem-edge/kernel-module-management/internal/utils"
+)
+
+const replaceKernelModuleScriptTemplate = `#!/bin/bash
+echo "before checking podman images"
+if podman images | grep $IMAGE | grep -q $TAG; then
+    echo "Image $IMAGE:$TAG found in the local registry, removing in-tree kernel module"
+    podman run --privileged --entrypoint modprobe $IMAGE:$TAG -rd /opt $KERNEL_MODULE
+    if [ $? -eq 0 ]; then
+            echo "Succesffully removed the in-tree kernel module $KERNEL_MODULE"
+    else
+            echo "failed to remove in-tree kernel module $KERNEL_MODULE"
+    fi
+    echo "Running container image to insert the oot kernel module $KERNEL_MODULE"
+    podman run --privileged --entrypoint modprobe $IMAGE:$TAG -d /opt $KERNEL_MODULE
+    echo "OOT kernel module $KERNEL_MODULE is inserted"
+else
+   echo "Image $IMAGE:$TAG is not present in local registry, will try after reboot"
+fi
+`
+const pullKernelModuleScriptTemplate = `#!/bin/bash
+if podman image list | grep $IMAGE | grep -q $TAG; then
```

Comment:
Can we use `podman images` or use `podman image list` in the previous script as well for consistency?

Reply:
fixed